server/src/main/java/org/elasticsearch/repositories/blobstore/ShardSnapshotTaskRunner.java

```diff
@@ -100,12 +99,12 @@ class FileSnapshotTask extends SnapshotTask {
 
         @Override
         public void doRun() {
-            ActionRunnable.run(fileSnapshotListener, () -> {
+            ActionListener.run(fileSnapshotListener, l -> {
                 FileInfo fileInfo = fileInfos.get();
                 if (fileInfo != null) {
                     fileSnapshotter.accept(context, fileInfo);
                 }
```

Comment:
The listener must be completed here:
```suggestion
                }
                l.onResponse(null)
```

Reply:
Aha of course. `ActionRunnable#run`, the gift that keeps on giving.